x-pack/plugins/infra/public/pages/metrics/hosts/components/tabs/metrics/metrics_grid.tsx

```diff
@@ -57,7 +57,7 @@ const CHARTS_IN_ORDER: Array<Pick<MetricChartProps, 'title' | 'type'> & { fullRo
   },
 ];
 
-export const MetricsGrid = () => {
+export const MetricsGrid = React.memo(() => {
```

Comment:
**note**: This component is memoized to prevent a full re-render when switching across tabs. It'll still rerender when required since it consumes contexts in the sub-tree

Reply:
I guess the charts will always re-render, right?

Reply:
Yep, it was re-rendering on each tab switch, but now it won't rerender unless the query, filters etc change :)